lib/ansible/modules/cloud/kubevirt/kubevirt_cdi_upload.py

```diff
@@ -160,10 +160,20 @@ def execute_module(self):
         headers = {'Authorization': "Bearer {0}".format(result['result']['status']['token'])}
         files = {'file': imgfile}
         url = "{0}/{1}/upload".format(upload_host, API)
-        requests.post(url, files=files, headers=headers, verify=upload_host_verify_ssl)
+        ret = requests.post(url, files=files, headers=headers, verify=upload_host_verify_ssl)
+
+        if ret.status_code != 200:
+            self.fail_request("Something went wrong while uploading data", method='POST', url=url,
+                              reason=ret.reason, status_code=ret.status_code)
 
         self.exit_json(changed=True)
 
+    def fail_request(self, msg, **kwargs):
```

Comment:
I am not sure why we need this extra method and the loop inside. Do you plan to reuse it somewhere?

Reply:
Yeah, this exact function is used elsewhere[1] and ideally it would be put in module_utils and reused. I decided not to try code refactors this late in the game, because time constraints. A little code duplication is fine assuming it can be cleaned up later.

[1] https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/clustering/k8s/k8s_auth.py#L317

Reply:
I would prefer to reuse

Reply:
Pushing PRs with code refactors that touch both us and `k8s` will take more than a few days (this isn't exactly high prio and time zones aren't on our side). And we have a few days. Reusing this function clearly indicates that I want to share the code in the feature. I just don't think that this is the time to do it.